spring-pulsar/build.gradle

```diff
@@ -15,6 +15,7 @@ dependencies {
 	api ('org.springframework.retry:spring-retry') {
 		exclude group: 'org.springframework'
 	}
+	api 'com.google.protobuf:protobuf-java:3.21.5'
```

Comment:
A couple of things on this new addition...

1. I was about to comment to suggest making this dep. optional as I thought that it would bring in `Guava`. **However**, I am glad that is not the case as it is only a `test` [dependency](https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java/3.21.5). I decided to keep this comment in order to say out loud that if we ever **have** to bring Guava in that we should shade it. It has caused me grief as a transitive dependency in the past. 




2. We have the versions managed via a Gradle Java platform in `spring-pulsar-dependencies`. Do you mind removing the version number here and then add a constraint for it in platform? This will look just like the one for [caffeine](https://github.com/spring-projects-experimental/spring-pulsar/blob/cc3842062e970ae4b64ade537d085bd0ec309ac9/spring-pulsar-dependencies/build.gradle#L39). I have been meaning to add an "Adding new dependencies" section for this in the [Working with the code Wiki](https://github.com/spring-projects-experimental/spring-pulsar/wiki/Working-with-the-Code) but have not gotten around to it yet.